File: recommendation_system/models/award.py
from datetime import datetime

from enums.genre import Genre


class Award:
    def __init__(self, id: int, title: str, date: datetime, category: Genre, description: str):
        self._id = id
        self._title = title
        self._date = date
        self._category = category
        self._description = description

    def set_title(self, title: str):
        self._title = title

    def get_title(self):
        return self._title

    def set_date(self, date: datetime):
        self._date = date

    def get_date(self):
        return self._date

    def set_category(self, category: Genre):
        self._category = category

    def get_category(self):
        return self._category

    def set_description(self, description: str):
        self._description = description

    def get_description(self):
        return self._description

File: movie_handler/src/main/java/com/example/model/Award.java
package com.example.model;

import com.example.model.enums.Genre;
import com.example.searchService.Search;
import javax.persistence.Entity;
import javax.persistence.Id;

import java.util.Date;
import java.util.UUID;

@Entity
public class Award {

    @Id
    private UUID id;
    private String title;
    private Date date;
    private Genre category;
    private String description;

    public Award(String title, Date date, Genre category, String description) {
        this.title = title;
        this.date = date;
        this.category = category;
        this.description = description;
    }

    public Award() {

    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Genre getCategory() {
        return category;
    }

    public void setCategory(Genre category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void accept(Search searchObject) {

    }
}
